lib/api/CDataFrameAnalysisConfigReader.cc

```diff
@@ -44,6 +44,12 @@ void CDataFrameAnalysisConfigReader::addParameter(const char* name,
     m_ParameterReaders.emplace_back(name, requirement, std::move(permittedValues));
 }
 
+void CDataFrameAnalysisConfigReader::addParameter(const std::string& name,
+                                                  ERequirement requirement,
+                                                  TStrIntMap permittedValues) {
+    addParameter(name.c_str(), requirement, std::move(permittedValues));
```

Comment:
Tom already said this on a previous PR, but as a followup it would be nice to remove the `const char*` overloads and make the `const std::string&` methods the only ones.  (Or at least make the `const std::string&` the ones that do the real work.)  We're losing the efficiency benefits of not having to call `strlen` lower down by converting to `const char*`.

Reply:
I will create an extra PR for this